ros2lifecycle/test/test_cli.py

```diff
@@ -0,0 +1,342 @@
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                '- configure [1]',
+                '\tStart: unconfigured',
+                '\tGoal: configuring',
+                '- shutdown [5]',
+                '\tStart: unconfigured',
+                '\tGoal: shuttingdown'
+            ],
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_list_all_lifecycle_node_transitions(self):
+        with self.launch_lifecycle_command(
+            arguments=['list', 'lc_node', '-a']
+        ) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=ALL_LIFECYCLE_NODE_TRANSITIONS,
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_list_all_lifecycle_hidden_node_transitions(self):
+        with self.launch_lifecycle_command(
+            arguments=['list', '_hidden_lc_node', '-a']
+        ) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=ALL_LIFECYCLE_NODE_TRANSITIONS,
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_list_lifecycle_nodes(self):
+        with self.launch_lifecycle_command(arguments=['nodes']) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['/lc_node'],
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_list_all_lifecycle_nodes(self):
+        with self.launch_lifecycle_command(arguments=['nodes', '-a']) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                '/_hidden_lc_node',
+                '/lc_node'
+            ],
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_count_all_lifecycle_nodes(self):
+        with self.launch_lifecycle_command(arguments=['nodes', '-a', '-c']) as lifecycle_command:
```

Comment:
Add a test counting without the -a flag checking that it doesn't count the hidden node

Reply:
Added in https://github.com/ros2/ros2cli/pull/391/commits/fc5977ccf05dac4a6800cda1175f95335b249ebb